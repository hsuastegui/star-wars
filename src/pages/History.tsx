import React from 'react'
import { Container, Card, Row, Anchor } from "../styles";

type props = {
    stack: stack
};

function History({ stack }: props) {
    return (
        <Container>
            <Anchor to="/">Play</Anchor>
            {stack.map((item) => {
                return (<Row>
                    <Card winner={item[0].value > item[1].value}>
                        <span>{item[0].name}</span>
                        <span>{item[0].value}</span>
                    </Card>
                    <Card winner={item[1].value > item[0].value}>
                        <span>{item[1].name}</span>
                        <span>{item[1].value}</span>
                    </Card>
                </Row>)
            })}
        </Container>
    )
}


export default History

