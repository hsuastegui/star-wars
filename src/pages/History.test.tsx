import React from "react";
import { shallow } from 'enzyme';
import History from './History';

const stack = [[{ "name": "R2-D2", "value": 96 }, { "name": "Bail Prestor Organa", "value": 191 }], [{ "name": "J-type diplomatic barge", "value": 0.7 }, { "name": "Naboo fighter", "value": 1 }], [{ "name": "TIE Advanced x1", "value": 1 }, { "name": "Star Destroyer", "value": 2 }], [{ "name": "Bail Prestor Organa", "value": 191 }, { "name": "Adi Gallia", "value": 184 }]]

describe("History", () => {
    it("renders", () => {
        const wrapper = shallow(<History stack={stack} />)
        expect(wrapper).toMatchSnapshot();
    })
})