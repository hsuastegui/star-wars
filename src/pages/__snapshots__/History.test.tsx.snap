// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History renders 1`] = `
<styled.div>
  <Styled(Link)
    to="/"
  >
    Play
  </Styled(Link)>
  <styled.div>
    <styled.div
      winner={false}
    >
      <span>
        R2-D2
      </span>
      <span>
        96
      </span>
    </styled.div>
    <styled.div
      winner={true}
    >
      <span>
        Bail Prestor Organa
      </span>
      <span>
        191
      </span>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      winner={false}
    >
      <span>
        J-type diplomatic barge
      </span>
      <span>
        0.7
      </span>
    </styled.div>
    <styled.div
      winner={true}
    >
      <span>
        Naboo fighter
      </span>
      <span>
        1
      </span>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      winner={false}
    >
      <span>
        TIE Advanced x1
      </span>
      <span>
        1
      </span>
    </styled.div>
    <styled.div
      winner={true}
    >
      <span>
        Star Destroyer
      </span>
      <span>
        2
      </span>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div
      winner={true}
    >
      <span>
        Bail Prestor Organa
      </span>
      <span>
        191
      </span>
    </styled.div>
    <styled.div
      winner={false}
    >
      <span>
        Adi Gallia
      </span>
      <span>
        184
      </span>
    </styled.div>
  </styled.div>
</styled.div>
`;
